### PR TITLE
Added (email) reminder support for calendar application

### PR DESCRIPTION
**Pull request by sureshprakash**

### CODE REVIEW

**Comment by georgehrke** [comment]
dup of #373

**Comment by georgehrke** [comment]
Sorry, this contains a change that is not present in #373, will close #373 instead

**Comment by georgehrke** [comment]
I am sorry, but this is not what is meant by https://github.com/owncloud/apps/issues/34. https://github.com/owncloud/apps/issues/34 is about the ability to set reminders with a custom time before the alarm for dedicated events.
This pull request implements a morning overview about what's on your schedule today. 
Sorry, but :-1: from my side.

**Comment by sureshprakash** [comment]
@georgehrke I agree with you! :)

**Comment by georgehrke** [comment]
Hey :)
sorry for the late response:

I see a few issues here:
- [ ] please add a configuration option (on a per user basis) so users have to actively opt in
- [ ] Most providers will kill a php script after more than 20 mails have been sent, the current solution doesn't scale
- [ ] add config option in admin settings for [this](https://github.com/owncloud/calendar/pull/374/files#diff-04c6e90faac2675aa89e2176d2eec7d8R35)
- [ ] revert your changes to readme.md, most of it is already documented in our admin doc
- [ ] please see our [coding style guidelines](http://doc.owncloud.org/server/6.0/developer_manual/app/general/codingguidelines.html) and fix what's wrong

File: lib/alarm.php [hunk 1]
```diff
@@ -5,9 +5,78 @@
...
+			
+			$separator = '~~~';
+		
+			$sql = 'SELECT `userid`, 
+					GROUP_CONCAT(`summary` ORDER BY `enddate`,`startdate` SEPARATOR \'' . $separator . '\') AS events, 
+					GROUP_CONCAT(`startdate` ORDER BY `enddate`,`startdate` SEPARATOR \'' . $separator . '\') AS startdate, 
+					GROUP_CONCAT(`enddate` ORDER BY `enddate`,`startdate` SEPARATOR \'' . $separator . '\') AS enddate 
+					FROM `oc_clndr_objects` AS objects 
+					JOIN `oc_clndr_calendars` AS calendars ON (`objects`.`calendarid`=`calendars`.`id`) 
+					WHERE CURDATE()>=DATE(`startdate`) AND CURDATE()<=DATE(`enddate`)
```
Comment (georgehrke) [inline lib/alarm.php:38]: There is not CURDATE() in SQLite

File: lib/alarm.php [hunk 2]
```diff
@@ -5,9 +5,78 @@
...
+				
+				$message = $tpl->fetchPage();
+				
+				/* $this->inc('event.reminder', 
+						array("event" => $event,
+							  "starting" => $starting,
+							  "ending" => $ending)
+						  ); */
+						  
+				OC_Mail::send(OC_Preferences::getValue($member, 'settings', 'email'), $member, $subject, $message, OC_Config::getValue('mail_smtpname', ''), 'Owncloud Event Reminder', true);
```
Comment (georgehrke) [inline lib/alarm.php:71]: Don't use private apis, please use the public apis (OCP...)

File: lib/alarm.php [hunk 3]
```diff
@@ -5,9 +5,78 @@
...
+		self::sendEmailRemainder();
+	}
+	
+	public function sendEmailRemainder()
+	{
+		try
+		{
+			$l = OC_L10N::get('calendar');
+			
+			$separator = '~~~';
```
Comment (georgehrke) [inline lib/alarm.php:30]: ???

File: lib/alarm.php [hunk 4]
```diff
@@ -5,9 +5,78 @@
...
+		
+			$sql = 'SELECT `userid`, 
+					GROUP_CONCAT(`summary` ORDER BY `enddate`,`startdate` SEPARATOR \'' . $separator . '\') AS events, 
+					GROUP_CONCAT(`startdate` ORDER BY `enddate`,`startdate` SEPARATOR \'' . $separator . '\') AS startdate, 
+					GROUP_CONCAT(`enddate` ORDER BY `enddate`,`startdate` SEPARATOR \'' . $separator . '\') AS enddate 
+					FROM `oc_clndr_objects` AS objects 
+					JOIN `oc_clndr_calendars` AS calendars ON (`objects`.`calendarid`=`calendars`.`id`) 
+					WHERE CURDATE()>=DATE(`startdate`) AND CURDATE()<=DATE(`enddate`) 
+					GROUP BY `userid` 
+					ORDER BY `enddate`,`startdate`';
```
Comment (georgehrke) [inline lib/alarm.php:40]: please use proper string-concatenation

**Comment by DeepDiver1975** [comment]
closing this - no feedback from contributor after initial review

**Pull request closed without merging**